### PR TITLE
Implement `symbol.prototype[ @@ToStringTag]`

### PR DESCRIPTION
**Pull request by moadmmh**
This Pull Request fixes/closes [#1231](https://github.com/boa-dev/boa/issues/1231).

It changes the following:

- implement  `symbol.prototype[ @@ToStringTag]`

### CODE REVIEW

File: boa/src/builtins/symbol/mod.rs [hunk 1]
```diff
@@ -91,6 +91,11 @@ impl BuiltIn for Symbol {
         )
         .callable(true)
         .constructable(false)
+        .property(
+            symbol_to_string_tag,
+            "Symbol",
```
Comment (Razican) [inline boa/src/builtins/symbol/mod.rs:96]: I would use `Self::NAME` here.

**Review by Razican — COMMENTED** [review]
It's looking good! Check my comment :)

**Comment by Razican** [comment]
### Test262 conformance changes:
  | Test result | master count | PR count | difference |
  | :---------: | :----------: | :------: | :--------: |
  | Total | 78,873 | 78,873 | 0 |
  | Passed | 26,665 | 26,667 | **+2** |
  | Ignored | 15,604 | 15,604 | 0 |
  | Failed | 36,604 | 36,602 | **-2** |
  | Panics | 14 | 14 | 0 |
  | Conformance | 33.81% | 33.81% | **+0.00%** |
  
  <details><summary><b>Fixed tests:</b></summary>
  ```
  test/built-ins/Symbol/prototype/Symbol.toStringTag.js [strict mode] (previously Failed)
  test/built-ins/Symbol/prototype/Symbol.toStringTag.js (previously Failed)
  ```
  </details>

File: boa/src/builtins/symbol/mod.rs (hunk 1 above)
Comment (moadmmh) [reply boa/src/builtins/symbol/mod.rs:96]: Yes I will change that 
Thanks for the review

**Review by Razican — APPROVED** [review]

**Review by RageKnify — APPROVED** [review]

**Pull request merged**